webapi/src/main/scala/org/knora/webapi/responders/v2/StandoffResponderV2.scala

```diff
@@ -0,0 +1,785 @@
+        case GetMappingRequestV2(mappingIri, userProfile) => future2Message(sender(), getMappingV2(mappingIri, userProfile), log)
+        case GetXSLTransformationRequestV2(xsltTextReprIri, userProfile) => future2Message(sender(), getXSLTransformation(xsltTextReprIri, userProfile), log)
+        case other => handleUnexpectedMessage(sender(), other, log, this.getClass.getName)
+    }
+
+    val xsltCacheName = "xsltCache"
+
+    /**
+      * If not already in the cache, retrieves a `knora-base:XSLTransformation` in the triplestore and requests the corresponding XSL transformation file from Sipi.
+      *
+      * @param xslTransformationIri The IRI of the resource representing the XSL Transformation (a [[OntologyConstants.KnoraBase.XSLTransformation]]).
+      * @param userProfile          The client making the request.
+      * @return a [[GetXSLTransformationResponseV1]].
+      */
+    private def getXSLTransformation(xslTransformationIri: IRI, userProfile: UserADM): Future[GetXSLTransformationResponseV2] = {
+
+        val xsltUrlFuture = for {
+
+            textRepresentationResponseV2: ReadResourcesSequenceV2 <- (responderManager ? ResourcesGetRequestV2(resourceIris = Vector(xslTransformationIri), requestingUser = userProfile)).mapTo[ReadResourcesSequenceV2]
+
+            xsltFileValue: TextFileValueContentV2 = textRepresentationResponseV2.resources.headOption match {
+                case Some(resource: ReadResourceV2) if resource.resourceClass == OntologyConstants.KnoraBase.XSLTransformation => resource.values.get(OntologyConstants.KnoraBase.HasTextFileValue) match {
+                    case Some(values: Seq[ReadValueV2]) if values.size == 1 => values.head match {
+                        case value: ReadValueV2 => value.valueContent match {
+                            case textRepr: TextFileValueContentV2 => textRepr
+                            case other => throw InconsistentTriplestoreDataException(s"${OntologyConstants.KnoraBase.XSLTransformation} $xslTransformationIri is supposed to have exactly one value of type ${OntologyConstants.KnoraBase.TextFileValue}")
+                        }
+                    }
+                    case None => throw InconsistentTriplestoreDataException(s"${OntologyConstants.KnoraBase.XSLTransformation} has no property ${OntologyConstants.KnoraBase.HasTextFileValue}")
+                }
+                case None => throw BadRequestException(s"Resource $xslTransformationIri is not a ${OntologyConstants.KnoraBase.XSLTransformation}")
+            }
+
+            // check if `xsltFileValue` represents an XSL transformation
+            _ = if (!(xsltFileValue.internalMimeType == "text/xml" && xsltFileValue.originalFilename.endsWith(".xsl"))) {
+                throw BadRequestException(s"$xslTransformationIri does not have a file value referring to an XSL transformation")
+            }
+
+            xsltUrl: String = s"${settings.internalSipiFileServerGetUrl}/${xsltFileValue.internalFilename}"
```

Comment:
@subotic Is this Sipi URL correct?

Using the v2 responder, I could get rid of the URL replacing operation that we needed for v1.

Reply:
@subotic Will this work on the production server? We had to do some replacement operations for v1.